src/Send/Memo.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {HelperText, TextInput} from '../components'
+
+export const maxMemoLength = 256
+
+type Props = {
+  memo: string
+  onChangeText: (memo: string) => void
+}
+
+export const Memo = ({onChangeText, memo}: Props) => {
+  const strings = useStrings()
+
+  const showError = memo.length > maxMemoLength
+  const multilineMax = memo.length > 50
+
+  return (
+    <View style={styles.container}>
+      <TextInput
+        value={memo}
+        onChangeText={(memo) => onChangeText(memo.trim())}
+        label={strings.label}
+        autoComplete={false}
+        testID="memoFieldInput"
+        errorText={showError ? 'error' : undefined} // to show the error styling
+        noHelper={true}
+        style={multilineMax && styles.input}
+        multiline={true}
+      />
+
+      <View style={styles.helper}>
+        <Message showError={showError} />
+
+        <LengthCounter memo={memo} showError={showError} />
+      </View>
+    </View>
+  )
+}
+
+const Message = ({showError}: {showError: boolean}) => {
+  const strings = useStrings()
+  return <HelperText type={showError ? 'error' : 'info'}>{showError ? strings.error : strings.warning}</HelperText>
+}
+
+const LengthCounter = ({memo, showError}: {memo: string; showError: boolean}) => {
+  return (
+    <HelperText type={showError ? 'error' : 'info'}>
+      <Text>{`${memo.length}/${maxMemoLength}`}</Text>
+    </HelperText>
+  )
+}
+
+const useStrings = () => {
+  const intl = useIntl()
+
+  return {
+    label: intl.formatMessage(messages.label),
+    warning: intl.formatMessage(messages.warning),
+    error: intl.formatMessage(messages.error),
+  }
+}
+
+export const messages = defineMessages({
+  label: {
+    id: 'components.send.memofield.label',
+    defaultMessage: '!!!Memo',
+  },
+  warning: {
+    id: 'components.send.memofield.warning',
+    defaultMessage: '!!!(Optional) Memo is stored locally',
+  },
+  error: {
+    id: 'components.send.memofield.error',
+    defaultMessage: '!!!Memo name is too long',
+  },
+})
+
+const styles = StyleSheet.create({
+  container: {
+    paddingBottom: 25,
+  },
+  helper: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  input: {
+    height: 95,
```

Comment:
shouldn't it be `maxHeight` ? otherwise the input will continue to expand

Reply:
Here you can read the requirements 

https://github.com/Emurgo/yoroi-mobile/pull/2324#discussion_r1105108493

1 line default until 3 lines maximum

Reply:
A check the video please

Reply:
different screen sizes, different text sizes
could lead to unpredictable behavior